### ISSUE TITLE
Shared vs not-shared?

### ISSUE THREAD

**Issue by teknowledgist** [MiKTeX/miktex#847]
Disclaimer:  I am a sysadmin supporting MiKTeX users, and I'm not much of a MiKTeX user myself.

I'm testing some deployment scenarios, and having trouble understanding with `miktexsetup_standalone.exe` what the what the `--shared=yes` (or `--shared`) switch does compared to the `--shared=no`.  In both cases (running as an Admin), MiKTeX is installed to `c:\Program Files\MiKTeX`, and appears in the Start Menu for all users and can be started without error by all users.  Also, with a `--verbose` switch, both versions return a `Setting config value: [Core]SharedSetup=1` line.

So, what is the difference?

**Comment by edocevoli** [comment]
`--shared=yes` is the default, if you run the installer with elevated privileges. This should be documented somewhere.

**Comment by teknowledgist** [comment]
What is the definition of a "shared installation"?  Alternatively, what is the default install location for a non-shared (i.e. this user only) installation?

I am installing elevated, but I feel like the `--shared=no` switch should take precedence.

**Comment by edocevoli** [comment]
A _shared installation_ is shared by all users.

The default location for a user-only installation is: `~/AppData/Local/Programs/MiKTeX`

**Comment by teknowledgist** [comment]
> A _shared installation_ is shared by all users.

Obviously.  😄 But what does that mean?  IOW, If an install is `--shared=no` and uses `c:\MiKTeX` as the install destination, how is that different than a `--shared=yes` install with the same destination?  Windows actually assigns less restrictive permissions to a user-created `c:\MiKTeX` than it does to an installed, `c:\Program Files\MiKTeX`.  (I assume the installer doesn't mess with inherited permissions on the folders it creates.)

If the only difference between shared and not-shared is the default install location, then that should probably be pointed out in the documentation.  

IMHO, if `--shared=yes` is specified, then by default the user directories should be set [as in the example](https://docs.miktex.org/manual/miktexsetup.html#id789544).  What would be the point of sharing otherwise?

> The default location for a user-only installation is: `~/AppData/Local/Programs/MiKTeX`
> 
> Yes, this must work, i.e., specifying `--shared=no` must install it for you only. No other user should have access to your private installation.

So, this is explicitly not working (at least when run elevated).  Can this issue act as a bug notice, or does it need to go somewhere else?

Thanks.

**Comment by edocevoli** [comment]
If `--shared=no` is specified, then configuration/data files will be created in the user home directory and not in `C:\ProgramData`.

**Comment by bazzilic** [comment]
Hi @edocevoli
So is this a confirmed bug? Will it be fixed?

**Comment by edocevoli** [comment]
I have not identified it as a bug. The issue is kept open as a reminder: the behavior should be documented.

**Comment by bazzilic** [comment]
If I pass `--shared=no` to an elevated installer where would it install the packages?

**Comment by edocevoli** [comment]
Right, there was something else 😄 

Running elevated with `--shared=no` must install in the home directory. Currently, the installer installs in `Program Files`. This is a bug.

**Comment by bazzilic** [comment]
Ok, cool, thanks! Looking forward to the fix!

**Comment by teknowledgist** [comment]
This is a bug, for sure.  Thanks for working toward a fix.

However, I opened this issue because I was trying to use `--shared=no` as a workaround solution for a problem that occurs for `--shared=yes`.  Most programs that have a "machine" (or "system") install work for all users and allow each user to have different settings and plugins.  I don't understand why MiKTeX doesn't work that way.  Sure, the "core" application files will require an admin to upgrade, but that's why sysadmins have admin rights and decide what software is installed/updated on the machines they manage.

If a shared installation is supposed to work similarly to "normal" software, consider what @bazzilic described to me:

>Main reason why private installation is preferred for single users is the on-the-fly package installation feature of miktex. In a private installation it just works, but in a machine-wide installation it only works if you use the miktex own tex editor (texworks is it?). If you use another editor, on-the-fly package installation just hangs the compilation (I assume because it can't show the UAC window and waits endlessly for user confirmation).

To me, that sounds like another bug.  

Thanks.

**Comment by edocevoli** [comment]
I am afraid that I don't understand what you mean. FYI, there are two settings which control unattended installation of packages:

```
[MPM]

	;; Install packages for all users.
	AutoAdmin = ?

	;; Install missing packages automatically (on-the-fly).
	AutoInstall = ?
```

So you can configure MiKTeX to

**Comment by bazzilic** [comment]
I assume that was "to auto-install packages in a per-user manner"? Yes, maybe I can configure it this way, I didn't know about this setting. However, IIRC, with default settings in a `--shared=yes` installation it auto-installs packages for all users, which in turn hangs, for example, when compilation is launched from VS Code.

**Comment by teknowledgist** [comment]
According to [the documentation](https://docs.miktex.org/manual/mpm.html#id779810), MPM runs in User Mode by default, which would imply that the `AutoAdmin` setting (where that is found) should not be enabled/true.  

@bazzilic: Any chance you run VS Code elevated (and have tried TeXworks unelevated)?

Maybe if MPM sees an elevated status it assumes Admin mode very much like the bug where the installer assumes `--shared=yes` when it sees elevated status?

**Comment by bazzilic** [comment]
I definitely don't run VS Code elevated :)

**Comment by edocevoli** [comment]
Please note that I fix a single bug (see comment above):

https://github.com/MiKTeX/miktex/issues/847#issuecomment-878152451

Please open a new issue, if you find that there are other bugs that needs to be fixed. Or if you have an enhancement suggestion.

**Comment by edocevoli** [comment]
This has been fixed in 21.7.
